src/Indices.jl

```diff
@@ -65,7 +65,15 @@ function Indices(iter)
         iter = collect(iter)
     end
 
-    return Indices{eltype(iter)}(iter)
+    # This is necessary to make map'ing across an empty dictionary work
+    # In both Julia 1.9 and 1.10 eltype(::Tuple{}) == Union{}
+    # but in Julia 1.10, this causes problems downstream when Indices{Union{}}
+    # is used. For an empty iterator, we actually don't know what the 
+    # True(tm) eltype is, so the top of the type hierarchy (Any) is 
+    # just as reasonable as the bottom (Union{})
+    I = typeof(iter) == Tuple{} ? Any : eltype(iter)
```

Comment:
> If you're constructing an Index from an empty tuple, you do not know what the original element type is, so Any IMHO makes as much sense as Union{}.

Small point: `Any` is technically correct (in a trivial sense), but less precise than `Union{}`. The empty union is the best possible (most accurate) type for the empty set of values. The possible breakage on the Julia side, however, is another thing...

Reply:
Small note: `typeof(iter) == Tuple{}` isn't very pretty/Julian. Here are two alternatives that are prettier in my opinion (although all  of them seem equally performant):

```julia
f(it) = (typeof(it) == Tuple{}) ? Any : eltype(it)
g(it) = (it == ()) ? Any : eltype(it)
h(it) = (it isa Tuple{}) ? Any : eltype(it)
```

Reply:
I believe `it === ()` or `it isa Tuple{}` are the correct forms. (Note: the latter form might invoke a slower subtyping algorithm when the LHS isn't inferred to a concrete type and/or the RHS is an abstract type; the former is always quite straightforward).